app/api/routes/product.py

```diff
@@ -1,148 +1,97 @@
+        products.delete_product(product_id, db)
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
 
-""" List Products"""
-@router.get("/products", status_code=200)
-async def list_products():
-    if not products_db:
-        return []
-    
-    products_list = []
-    for product_id, product_data in products_db.items():
-        product_dict = {
-            "id": product_id,
-            "name": product_data["name"],
-            "price": product_data["price"],
-            "description": product_data.get("description", ""), 
-            "stock": product_data.get("stock", 0),  
-            "isAvailable": product_data.get("isAvailable", True),  
-            "created_at": product_data["created_at"]
-        }
-        products_list.append(product_dict)
-    return products_list
-    
-""" Search for Product"""
-@router.get("/products/search",status_code=200,response_model=dict)
-async def search_products(product: ProductSearchParams):
-    name = product.name
-    min_price = product.min_price
-    max_price = product.max_price
-    isAvailable = product.isAvailable
-    page = product.page
-    page_size = product.page_size
-    sort_by = product.sort_by
-    sort_order = product.sort_order
-    
-    filtered_products = []
-# Filling the array 
-    for product_id, product_data in products_db.items():
-        if name and name.lower() not in product_data["name"].lower():
-            continue
-        if min_price and product_data["price"] < min_price:
-            continue
-        if max_price and product_data["price"] > max_price:
-            continue
-        if isAvailable is not None and product_data["isAvailable"] != isAvailable:
-            continue
-        filtered_products.append({
-            "id": product_id,
-            "name": product_data["name"],
-            "price": product_data["price"],
-            "stock": product_data["stock"],
-            "isAvailable": product_data["isAvailable"],
-            "created_at": product_data["created_at"]
-        })
-        
-    if not filtered_products:
-        return []
-    
-    if sort_order == "asc":
-        filtered_products.sort(key=lambda x: x[sort_by])
-    elif sort_order == "desc":
-        filtered_products.sort(key=lambda x: x[sort_by], reverse=True)
-    total_products = len(filtered_products)
-    total_pages = (total_products + page_size - 1) // page_size
-    start = (page - 1) * page_size
-    end = start + page_size
-    paginated_products = filtered_products[start:end]
-    
-    return {
-        "page": page,
-        "total_pages": total_pages,
-        "products_per_page": page_size,
-        "total_products": total_products,
-        "products": paginated_products
-    }
\ No newline at end of file
+# Endpoint to list all products
+@router.get("/products", response_model=List[schemas.Product], status_code=status.HTTP_200_OK)
+def list_products_endpoint(
```

Comment:
You can include pagination here, but without any filters. The filters should only be applied in `products/search`.

Reply:
Done